lib/evaluateFunctions.js

```diff
@@ -96,4 +99,389 @@ function absoluteValue(node) {
+  const radicandNode = node.args[0];
+  const rootNode = node.args.length === 2 ? node.args[1] : NodeCreator.constant(2);
+
+  const children = [];
+  for (let i = 0; i < radicandNode.args.length; i++) {
+    const child = radicandNode.args[i];
+    children.push(NodeCreator.nthRoot(child, rootNode));
+  }
+
+  newNode = NodeCreator.operator('*', children);
+  return NodeStatus.nodeChanged(
+    MathChangeTypes.DISTRIBUTE_NTH_ROOT, node, newNode);
+}
+
+// Given a multiplication node of nthRoots (with the same root)
+// will evaluate the nthRoot of each Node as a substep
+// e.g. nthRoot(2) * nthRoot(x^2) -> nthRoot(2) * x
+function evaluateNthRootForChildren(node) {
+  let newNode = clone(node, false);
+
+  const substeps = [];
+  for (let i = 0; i < newNode.args.length; i++) {
+    const child = newNode.args[i];
+    const childNodeStatus = nthRoot(child);
+    if (childNodeStatus.hasChanged()) {
+      newNode.args[i] = childNodeStatus.newNode;
+      substeps.push(NodeStatus.childChanged(newNode, childNodeStatus, i));
+    }
+  }
+
+  if (substeps.length > 0) {
+    return NodeStatus.nodeChanged(
+      MathChangeTypes.EVALUATE_DISTRIBUTED_NTH_ROOT, node, newNode, true, substeps);
+  }
+
+  return NodeStatus.noChange(node);
+}
+
+// Given a multiplication nodes, including nthRoots, will combine the nodes
+// with the same radicand as a step
+// e.g. 2 * nthRoot(2) * nthRoot(x) -> 2 * nthRoot(2 * x)
+function combineRoots(node) {
+  let newNode = clone(node, false);
+  const rootNode = node.args.length === 2 ? node.args[1] : NodeCreator.constant(2);
+
+  const children = [];
+  const radicandArgs = [];
+  for (let i = 0; i < newNode.args.length; i++) {
+    const child = newNode.args[i];
+    if (NodeType.isFunction(child, 'nthRoot') && child.args[1].equals(rootNode)) {
+      radicandArgs.push(child.args[0]);
+    }
+    else {
+      children.push(child);
+    }
+  }
+
+  // if there are no items moved out of the root, then nothing has changed
+  if (children.length > 0) {
+    if (radicandArgs.length > 0) {
+      const radicandNode = radicandArgs.length === 1 ?
+        radicandArgs[0] : NodeCreator.operator('*', radicandArgs);
+      children.push(NodeCreator.nthRoot(radicandNode, rootNode));
+    }
+
+    newNode = NodeCreator.operator('*', children);
+    return NodeStatus.nodeChanged(
+      MathChangeTypes.COMBINE_UNDER_ROOT, node, newNode);
+  }
+
+  return NodeStatus.noChange(node);
+}
+
+// Returns the nthRoot evaluated on a constant node
+// Potentially factors the constant node into primes, and calls
+// nthRootMultiplication on the new nthRoot
+function nthRootConstant(node) {
+  let newNode = clone(node, false);
+  const radicandNode = node.args[0];
+  const rootNode = node.args.length === 2 ? node.args[1] : NodeCreator.constant(2);
+
+  if (Negative.isNegative(radicandNode)) {
+    return NodeStatus.noChange(node);
+  }
+  else if (!NodeType.isConstant(rootNode) || Negative.isNegative(rootNode)) {
```

Comment:
sqrt(-36) is still 6 * root(-1), so I think negative should be allowed and the -1 can be part of the prime factoring

also you should probably check integer and not just constant